host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_test10_ZoomModeAndVerify.py

```diff
@@ -0,0 +1,121 @@
+
+from L3_TestCases.dsHdmiIn.dsHdmiInHelperClass import dsHdmiInHelperClass
+
+class dsHdmiIn_test10_ZoomModeAndVerify(dsHdmiInHelperClass):
+    """
+    A test class for verifying the Zoom Mode of HDMI inputs.
+
+    This class extends `dsHdmiInHelperClass` and provides functionality
+    to test the zoom modes of HDMI inputs on the device.
+    """
+
+    def __init__(self):
+        """
+        Initializes the test10_ZoomModeAndVerify test .
+
+        Args:
+            None.
+        """
+        self.testName  = "test10_ZoomModeAndVerify"
+        super().__init__(self.testName, '1')
+
+    #TODO: Current version supports only manual verification.
+    def CheckDeviceStatusAndVerifyZoomMode(self, Manual=False, port_type:str=0,videoZoomMode:str=0):
+        """
+        Verifies whether the particular zoom mode is selected or not.
+
+        Args:
+            manual (bool, optional): Indicates if manual verification is needed.
+            port_type (str): The type of the port being verified.
+            video_zoom_mode (bool): Indicates if zoom mode verification is required.
+
+        Returns:
+            bool: True if verification is successful, otherwise False.
+        """
+        if Manual == True and videoZoomMode == False:
+            self.testUserResponse.getUserYN(f'Please connect the {port_type} and press Enter:')
+            time.sleep(3)
+            return self.testUserResponse.getUserYN(f'Is HDMI device connected and display is ON {port_type}? Y/N:')
+        elif Manual == True and videoZoomMode == True:
+            return self.testUserResponse.getUserYN(f'Verify Zoom Mode selected on port {port_type} and press Enter')
+        else :
+            #TODO: Add automation verification methods
+            return False
+
+    def testFunction(self):
+        """
+        The main test function to verify the Zoom Mode of HDMI input devices.
+
+        This function:
+        - Initializes the HDMI input module.
+        - Loops through each supported HDMI port.
+        - Selects the port and verifies the zoom modes.
+        """
+
+        self.log.testStart(self.testName, '1')
+
+        # Initialize the dsHDMIIn module
+        self.testdsHdmiIn.initialise()
+        result = True
+
+        # Loop through the supported HdmiIn ports
+        for port in self.testdsHdmiIn.getSupportedPorts():
+            self.log.stepStart(f'Select {port} Port')
+            self.log.step(f'Select {port} Port')
+
+            # Check the HdmiIn device connected to is active
+            result = self.CheckDeviceStatusAndVerifyZoomMode(True,port,False)
+            if not result:
+                self.testdsHdmiIn.selectHDMIInPort(port, audMix=0, videoPlane=0, topmost=1)
+                time.sleep(5)
+                self.log.step(f'Port Selected {port}')
+
+            #get the list of Zoom Modes
+            zoomModeList = self.testdsHdmiIn.getVideoZoomModeList()
+
+            for zoomModeIndex in range (1,len(zoomModeList)):
```

Comment:
zoomModeList is a list, it can be used directly in loop
for zoomMode in zoomModeList: